programs/survival/silk_breakables.sc

```diff
@@ -0,0 +1,238 @@
+// sound to play when broken
+global_blocks = {
+    'bedrock' -> [75, 'netherite_pickaxe', [null, null], false, 'stone' ],
+    'budding_amethyst' -> [1, ['diamond_pickaxe', 'netherite_pickaxe'], ['budding_amethyst', null], false, 'amethyst_block' ],
+    'deepslate' -> [2, 'netherite_pickaxe', ['deepslate', 'cobbled_deepslate'], false, 'deepslate' ],
+    'end_portal_frame' -> [25, ['diamond_pickaxe', 'netherite_pickaxe'], ['end_portal_frame', null], false, 'stone' ],
+    'spawner' -> [1, ['diamond_pickaxe', 'netherite_pickaxe'], ['spawner', null], true, 'metal' ],
+};
+
+global_tool_speeds = {
+    'wooden_axe' -> 2,
+    'stone_axe' -> 4,
+    'iron_axe' -> 6,
+    'diamond_axe' -> 8,
+    'netherite_axe' -> 9,
+    'golden_axe' -> 12,
+    'wooden_pickaxe' -> 2,
+    'stone_pickaxe' -> 4,
+    'iron_pickaxe' -> 6,
+    'diamond_pickaxe' -> 8,
+    'netherite_pickaxe' -> 9,
+    'golden_pickaxe' -> 12,
+    'wooden_shovel' -> 2,
+    'stone_shovel' -> 4,
+    'iron_shovel' -> 6,
+    'diamond_shovel' -> 8,
+    'netherite_shovel' -> 9,
+    'golden_shovel' -> 12,
+    'wooden_hoe' -> 2,
+    'stone_hoe' -> 4,
+    'iron_hoe' -> 6,
+    'diamond_hoe' -> 8,
+    'netherite_hoe' -> 9,
+    'golden_hoe' -> 12,
+};
+
+__check_tool (player, tools) -> (
+    [tool, count, nbt] = player ~ 'holds' || ['None', 0, null];
+
+    type(tools) !='list' && (tools = l(tools));
+    for (tools,
+    	str(tool) == _ && (valid_tool = true);
+    );
+
+    return(valid_tool);
+);
+
+__get_enchantment_level(tool_nbt, enchantment) -> (
+    level = 0;
+	if ((enchantments = get(tool_nbt,'Enchantments[]')),
+		if (type(enchantments)!='list', enchantments = l(enchantments));
+		for (enchantments,
+			if ( get(_,'id') == 'minecraft:'+enchantment,
+				level = max(level, get(_,'lvl'))
+			)
+		)
+	);
+	level
+);
+
+__calculate_step (player, hardness) -> (
+
+    [tool, count, nbt] = player ~ 'holds' || ['None', 0, null];
+
+    speedMultiplier = global_tool_speeds:str(tool);
+
+    efficiency_level = __get_enchantment_level(nbt, 'efficiency');
+    efficiency_level && (
+        speedMultiplier = speedMultiplier + 1 + (efficiency_level * efficiency_level);
+    );
+
+    haste_level = query(player,'effect','haste'):0;
+    haste_level && (
+        speedMultiplier = speedMultiplier * (1 + (0.2 * (haste_level + 1)))
+    );
+
+    mining_fatigue_level = query(player,'effect','mining_fatigue'):0 || -1;
+    if (mining_fatigue_level == 0,
+        (
+            speedMultiplier = speedMultiplier * 0.3;
+        ),
+        // else if
+        mining_fatigue_level == 1,
+        (
+            speedMultiplier = speedMultiplier * 0.09;
+        ),
+        // else if
+        mining_fatigue_level == 2,
+        (
+            speedMultiplier = speedMultiplier * 0.0027;
+        ),
+        // else if
+        mining_fatigue_level >= 3,
+        (
+            speedMultiplier = speedMultiplier * 0.00081;
+        )
+    );

```

Comment:
I like the `else if` comments btw.

I would tend to attempt to use a map/hash for this. 
Tho I see that could also get complicated.

Reply:
thanks i needed to find some way to make it readable as i've been working on this one off and on over a couple of months and since my day to day is c++ & python it takes a bit for me to switch to scarpets more pure functional syntax w/o comments like that